src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -177,7 +177,7 @@ public override Message VisitMessage(Message node)
                 }
 
                 if (node.Text == null &&
-                    _run.Tool?.GlobalMessageStrings?.TryGetValue(node.MessageId, out formatString) == true)
+                    _run.Tool.Driver.GlobalMessageStrings?.TryGetValue(node.MessageId, out formatString) == true)
```

Comment:
>. [](start = 29, length = 1)

Why did you remove the null conditional operator?

Reply:
check line 168, where it didn't exist. Tool.Driver is a required object. we could put an arg check at the top of the method

---
In reply to: [256108174](https://github.com/Microsoft/sarif-sdk/pull/1277#discussion_r256108174) [](ancestors = 256108174)